osu.Game/Rulesets/Mods/ClockWithMods.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Linq;
+
+namespace osu.Game.Rulesets.Mods
+{
+    public class ClockWithMods
+    {
+        protected IApplicableToRate? Mod;
+
+        public ClockWithMods(Mod[] mods)
+        {
+            Mod = mods.OfType<IApplicableToRate>().SingleOrDefault();
+        }
+
+        public double ApplyToRate(double time, double rate = 1) => Mod?.ApplyToRate(time) ?? 1;
+
+        public double GetAverageRate() => Mod?.GetAverageRate() ?? 1;
+
+        public double GetTimeAt(double time) => Mod?.GetTimeAt(time) ?? time;
```

Comment:
I'm not sure on this implementation - right now DT and WU/WD are incompatible but I believe there used to be a point in time in which they were compatible, and technically speaking I don't think they need to be incompatible (though I could be wrong on this).

So that being said, is it possible to instead have `IApplicableToRate` return a rate multiplier at the given time and do the summation ourselves?

Reply:
As a follow up, it looks like it should be possible to use `ApplyToRate()` as-is, which is used in other places in a very similar way to what I'm describing here:

https://github.com/ppy/osu/blob/b4a3f872ee73c40a8f25356efca2eae2ba382e3b/osu.Game.Rulesets.Osu/Replays/OsuAutoGeneratorBase.cs#L68-L73